src/rpc/methods/f3.rs

```diff
@@ -541,6 +564,69 @@ impl RpcMethod<1> for F3GetF3PowerTable {
     }
 }
 
+/// F3Participate should be called by a storage provider to participate in signing F3 consensus.
+/// Calling this API gives the lotus node a lease to sign in F3 on behalf of given SP.
+/// The lease should be active only on one node. The lease will expire at the newLeaseExpiration.
+/// To continue participating in F3 with the given node, call F3Participate again before the newLeaseExpiration time.
+/// newLeaseExpiration cannot be further than 5 minutes in the future.
+/// It is recommended to call F3Participate every 60 seconds with newLeaseExpiration set 2min into the future.
+/// The oldLeaseExpiration has to be set to newLeaseExpiration of the last successful call.
+/// For the first call to F3Participate, set the oldLeaseExpiration to zero value/time in the past.
+/// F3Participate will return true if the lease was accepted. The minerID has to be the ID address of the miner.
+pub enum F3Participate {}
+impl RpcMethod<3> for F3Participate {
+    const NAME: &'static str = "Filecoin.F3Participate";
+    const PARAM_NAMES: [&'static str; 3] = [
+        "miner_address",
+        "new_lease_expiration",
+        "old_lease_expiration",
+    ];
+    const API_PATHS: ApiPaths = ApiPaths::V1;
+    const PERMISSION: Permission = Permission::Read;
+
+    type Params = (
+        Address,
+        chrono::DateTime<chrono::Utc>,
+        chrono::DateTime<chrono::Utc>,
+    );
+    type Ok = bool;
+
+    async fn handle(
+        _: Ctx<impl Blockstore>,
+        (miner, new_lease_expiration, old_lease_expiration): Self::Params,
+    ) -> Result<Self::Ok, ServerError> {
+        if new_lease_expiration > chrono::Utc::now() + chrono::Duration::minutes(5) {
+            return Err(anyhow::anyhow!("F3 participation lease cannot be over 5 mins").into());
+        } else if new_lease_expiration < chrono::Utc::now() {
+            return Err(anyhow::anyhow!("F3 participation lease is in the past").into());
+        }
+        let id = miner.id()?;
+        // if the old lease is expired just insert a new one
+        if old_lease_expiration < chrono::Utc::now() {
```

Comment:
The logic matches lotus but it seems any old value can bypass all checks

Reply:
https://github.com/filecoin-project/lotus/blob/v1.29.0-rc1/chain/lf3/leasemanager.go#L33